### DISCUSSION TITLE
Recommended approach to implementing 3D interactive stimuli

### DISCUSSION THREAD

**Discussion by RylanSchaeffer** [jspsych/jsPsych#2467] [Q&A]
I want to run an experiment in which, on each trial, I ask participants to rotate a 3D object by dragging their mouse. Is there an easy way to do this in jsPsych?

If so, where can I learn more?

If not, how would someone with more experience (I have none) recommend I implement this?

**Comment by jodeleeuw** [comment]
Hi @RylanSchaeffer,

There's no plugin included with jsPsych that will let you do this super easily. The various `canvas-` plugins would let you render 3D objects (e.g., see [this MDN example of rendering a 3D object onto a `<canvas>` element). The mouse dragging, however, is not part of the plugin specs.

If this were a task I needed to implement I would probably look for a JavaScript library that can do this kind of thing already and then build a plugin that uses that library. Perhaps something like [three.js](https://threejs.org/), but I'm not sure what the best options are.
